File: src/domain/master/value_objects/severity.py
from enum import IntEnum


class Severity(IntEnum):
    """Severity levels for monitoring events"""

    UNKNOWN = 0
    LOW = 1
    MEDIUM = 2
    HIGH = 3
    CRITICAL = 4

    @classmethod
    def from_string(cls, value: str) -> "Severity":
        """Create Severity from string representation"""
        mapping = {
            "unknown": cls.UNKNOWN,
            "low": cls.LOW,
            "medium": cls.MEDIUM,
            "high": cls.HIGH,
            "critical": cls.CRITICAL,
        }
        return mapping.get(value.lower(), cls.UNKNOWN)

    def is_critical(self) -> bool:
        """Check if severity is critical or high"""
        return self >= self.HIGH

    def requires_immediate_action(self) -> bool:
        """Check if severity requires immediate action"""
        return self == self.CRITICAL

File: frontend/src/core/enums/severity.ts
/**
 * Event severity levels
 */
export enum Severity {
  Critical = 'critical',
  High = 'high',
  Medium = 'medium',
  Low = 'low',
  Unknown = 'unknown'
}

export const SeverityLabels: Record<Severity, string> = {
  [Severity.Critical]: 'Critical',
  [Severity.High]: 'High',
  [Severity.Medium]: 'Medium',
  [Severity.Low]: 'Low',
  [Severity.Unknown]: 'Unknown'
}

export const SeverityColors: Record<Severity, { bg: string; text: string; icon: string }> = {
  [Severity.Critical]: { bg: 'bg-red-100', text: 'text-red-800', icon: '🔴' },
  [Severity.High]: { bg: 'bg-orange-100', text: 'text-orange-800', icon: '🟠' },
  [Severity.Medium]: { bg: 'bg-blue-100', text: 'text-blue-800', icon: '🔵' },
  [Severity.Low]: { bg: 'bg-green-100', text: 'text-green-800', icon: '🟢' },
  [Severity.Unknown]: { bg: 'bg-gray-100', text: 'text-gray-800', icon: '⚪' }
}
